fdbsyncd/fdbsync.cpp

```diff
@@ -0,0 +1,603 @@
+    m_mac[key].ifname = intf_name;
+
+    macUpdateVxlanDB(key);
+    return;
+}
+
+void FdbSync::macDelVxlan(string key)
+{
+    if (m_mac.find(key) != m_mac.end())
+    {
+        SWSS_LOG_INFO("DEL_KEY %s vtep:%s type:%d", key.c_str(), m_mac[key].vtep.c_str(), m_mac[key].type);
+        macDelVxlanDB(key);
+        m_mac.erase(key);
+    }
+    return;
+}
+
+void FdbSync::onMsgNbr(int nlmsg_type, struct nl_object *obj)
+{
+    char macStr[MAX_ADDR_SIZE + 1] = {0};
+    struct rtnl_neigh *neigh = (struct rtnl_neigh *)obj;
+    struct in_addr vtep = {0};
+    int vlan = 0, ifindex = 0;
+    uint32_t vni = 0;
+    nl_addr *vtep_addr;
+    string ifname;
+    string key;
+    bool delete_key = false;
+    size_t str_loc = string::npos;
+    uint32_t type = 0;
+    bool isNotVxlanLink = false;
+    string vlan_id = "";
+    bool isIntfExist = false;
+
+    if ((nlmsg_type != RTM_NEWNEIGH) && (nlmsg_type != RTM_GETNEIGH) &&
+        (nlmsg_type != RTM_DELNEIGH))
+        return;
+
+    /* Only MAC route is to be supported */
+    if (rtnl_neigh_get_family(neigh) != AF_BRIDGE)
+    {
+        return;
+    }
+    ifindex = rtnl_neigh_get_ifindex(neigh);
+    if (intf_info.find(ifindex) != intf_info.end())
+    {
+        isIntfExist = true;
+        ifname = intf_info[ifindex].ifname;
+    }
+
+    nl_addr2str(rtnl_neigh_get_lladdr(neigh), macStr, MAX_ADDR_SIZE);
+
+    if (isIntfExist == false)
+    {
+        /* In Normal run kernel should not ageout this mac */
+        isNotVxlanLink = true;
+
+        /* When the mac is not learned on the vxlan interface then check if we need to refresh the route */
+        if (nlmsg_type != RTM_DELNEIGH)
+            return;
+    }
+    else
+    {
+        /* If this is for vnet bridge vxlan interface, then return */
+        if (ifname.find(VXLAN_BR_IF_NAME_PREFIX) != string::npos)
+            return;
+
+        /* VxLan netdevice should be in <name>-<vlan-id> format */
+        str_loc = ifname.find("-");
+        if (str_loc == string::npos)
+            return;
+
+        vlan_id = "Vlan" + ifname.substr(str_loc+1,  std::string::npos);
+        vni = intf_info[ifindex].vni;
+    }
+
+
+    if (isNotVxlanLink)
+    {
+        vlan = rtnl_neigh_get_vlan(neigh);
+        macRefreshStateDB(vlan, macStr);
+        return;
+    }
+
+    vtep_addr = rtnl_neigh_get_dst(neigh);
+    if (vtep_addr == NULL)
+    {
+        return;
+    }
+    else
+    {
+        /* Currently we only support ipv4 tunnel endpoints */
+        vtep.s_addr = *(uint32_t *)nl_addr_get_binary_addr(vtep_addr);
+        SWSS_LOG_INFO("Tunnel IP %s Int%d", inet_ntoa(vtep), *(uint32_t *)nl_addr_get_binary_addr(vtep_addr));
+    }
+
+    int state = rtnl_neigh_get_state(neigh);
+    if ((nlmsg_type == RTM_DELNEIGH) || (state == NUD_INCOMPLETE) ||
+        (state == NUD_FAILED))
+    {
+        delete_key = true;
+    }
+
+    if (state & NUD_NOARP)
+    {
+        /* This is a static route */
+        type = true;

```

Comment:
type is defined as uint32 but used as bool. Do you want to use string itself to mention "static"